ovirt_imageio/client/_tool.py

```diff
@@ -27,7 +34,23 @@ def main():
         format="%(asctime)s %(levelname)-7s (%(threadName)s) [%(name)s] "
                "%(message)s")
 
-    # XXX Setup signal handlers.
-    # XXX Handle commands errors.
-
-    args.command(args)
+    signal.signal(signal.SIGINT, terminate)
+    signal.signal(signal.SIGTERM, terminate)
+
+    try:
+        args.command(args)
+    except Exception:
```

Comment:
Catching `Exception` is necessary, but there are cases where it may be nice to catch specific exceptions.

`ovirtsdk4.Error` is a good candidate to be catch separately. We could do `log.error` with the exception string instead of `log.exception`, so we skip the Traceback. Traceback is useful for developers, but probably just noisy for users.

Reply:
I don't want to depend on sdk types here. If we think that dome errors need special handling, we need to wrap the sdk error with tool specific errors.

We can add some standard errors like client._errors.NoSuchDisk, client._errors.NoSuchStorageDomain, client._errors.NoSuchImage and show a traceback only for unexpected errors.

Lets open an issue for this for now. First version always logging tracebacks is OK.

Reply:
Opened #112 to track this. This may not be easy to fix in all cases.